### DISCUSSION TITLE
~/.sumi/extensions目录能否修改成其他目录

### DISCUSSION THREAD

**Discussion by 10000l** [opensumi/core#1915] [Q&A]
请问web版扩展安装目录~/.sumi/extensions，能否自己定义

**Comment by erha19** [comment]
@linxiaolx-lx 可以根据 [集成配置文档](https://opensumi.com/zh/docs/integrate/universal-integrate-case/custom-config#browser-%E9%85%8D%E7%BD%AE) 中的 `extensionDir` 属性配置一下期望的路径。

以 ide-startup 示例项目为例子，代码见：https://github.com/opensumi/ide-startup/blob/master/src/browser/render-app.ts#L22

**Reply by 10000l** [reply]
<img width="282" alt="image" src="https://user-images.githubusercontent.com/60864454/199871611-eb79da07-3ea7-4667-a2bf-52337b5f3dba.png">
<img width="748" alt="image" src="https://user-images.githubusercontent.com/60864454/199871776-7289619d-ac4f-4905-b971-5bde50fa0cd6.png">

修改了extensionDir参数之后预置的插件会安装到设置目录但是扩展市场开发自主安装的还是会安装到～/.sumi/extensions目录中

**Reply by erha19** [reply]
“扩展市场开发自主安装” 这个是啥？

**Reply by 10000l** [reply]
> “扩展市场开发自主安装” 这个是啥？

这个没问题了，我原本是想修改一下扩展市场中扩展安装的目录，这个不是修改前端的，是修改node的配置[mark](https://github.com/opensumi/ide-startup/blob/master/src/node/start-server.ts#L33)

**Comment by Aaaaash** [comment]
https://opensumi.com/zh/docs/integrate/universal-integrate-case/custom-config
